### PR TITLE
oop adapter connection auth config

### PR DESCRIPTION
**Pull request by bianpengyuan**
https://github.com/istio/istio/issues/7637

@mandarjog This is a draft and I hope to get your early review. I am going to make a prototype with this config to make sure it works as expected. I've considered to reuse the [TLSSettings](https://istio.io/docs/reference/config/istio.networking.v1alpha3/#TLSSettings), but it does not quite match  oop adapter auth.

### CODE REVIEW

**Comment by bianpengyuan** [comment]
cc @wattli

**Comment by bianpengyuan** [comment]
@mandarjog I updated the config proto. ptal.

File: policy/v1beta1/cfg.proto [hunk 1]
```diff
@@ -307,5 +307,100 @@ message Connection {
...
+
+  // AuthHeader specifies how to pass access token with authorization header.
+  enum AuthHeader {
+    // Access token is passed in authorization header as what it is
+    // (authorization: some-token).
+    PLAIN = 0;
+
+    // Access token is passed to adapter as basic token (i.e. authorization:
+    // basic some-token).
+    BASIC = 1;
```
Comment (mandarjog) [inline policy/v1beta1/cfg.proto]: Lets get rid of Basic auth. we should keep api-key auth and oauth.

Comment (bianpengyuan) [reply policy/v1beta1/cfg.proto]: Done.

**Comment by bianpengyuan** [comment]
@mandarjog PTAL and let's check this in and start implementation in istio repo?

**Comment by bianpengyuan** [comment]
@mandarjog gentle ping.

File: policy/v1beta1/cfg.proto [hunk 2]
```diff
@@ -307,5 +307,96 @@ message Connection {
...
+    // key, etc).
+    string token_path = 2;
+
+    // Oauth config to fetch access token from auth provider.
+    OAuth oauth = 3;
+  }
+
+  // AuthHeader specifies how to pass access token with authorization header.
+  enum AuthHeader {
+    // Access token is passed in authorization header as what it is
```
Comment (brian-avery) [inline policy/v1beta1/cfg.proto]: Should this also support Basic Auth headers?  Basic authentication should be of the format: Authorization: Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==

**Comment by bianpengyuan** [comment]
@mandarjog just rebased to 1.1. Ptal.

**Comment by bianpengyuan** [comment]
@mandarjog Any chance to take a look and check this in? :)

**Comment by mandarjog** [comment]
The design has been discussed in [Authentication for Mixer-grpc Adapters](https://docs.google.com/document/d/1lp96Ib_mMNZAMnsf8VTBAMOpwbmOrjBKgaq1XN26Ub8/edit#) design doc.

**Pull request merged**